plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -108,25 +112,37 @@ public CompletionStage<PutDataObjectResponse> putDataObjectAsync(PutDataObjectRe
         final String id = request.id() != null ? request.id() : UUID.randomUUID().toString();
         final String tenantId = request.tenantId() != null ? request.tenantId() : DEFAULT_TENANT;
         final String tableName = getTableName(request.index());
+        final GetItemRequest getItemRequest = getGetItemRequest(request.tenantId(), request.id(), request.index());
+
         return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<PutDataObjectResponse>) () -> {
             try {
+                GetItemResponse getItemResponse = dynamoDbClient.getItem(getItemRequest);
+                Long sequenceNumber = 1L;
+                if (getItemResponse != null && getItemResponse.item() != null && getItemResponse.item().containsKey(SEQ_NO_KEY)) {
+                    sequenceNumber = Long.parseLong(getItemResponse.item().get(SEQ_NO_KEY).n()) + 1;
+                }
```

Comment:
(Readability) Might be nice to excerpt these lines into a local private method `initOrIncrementSeqNo()`